### PR TITLE
refactor(sdk-coin-atom): only decode sequence from raw atom tx

### PR DESCRIPTION
**Pull request by michael-gallegos-bg**
We found that we no longer need the accountNumber or the chainId to properly construct a raw tx.

Also adds more testing regarding transactionBuilder and factory building.

TICKET: BG-67573

### CODE REVIEW

File: modules/sdk-coin-atom/src/lib/transaction.ts [hunk 1]
```diff
@@ -125,6 +125,8 @@ export class Transaction extends BaseTransaction {
   }
 
   serialize(): string {
+    // TODO - Check if we can get the pub key from the encoded/decoded tx data
```
Comment (jicli) [inline modules/sdk-coin-atom/src/lib/transaction.ts:128]: add a ticket for it plz

Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transaction.ts:128]: this is a draft PR, consulting with cosmos team on how we wish to proceed

**Comment by michael-gallegos-bg** [comment]
From @jicli : "but we just changed to sequence id?"
Yes, so we used to implement this using the Stargate Client from the Cosmos JS SDK, which requires an account number/chain ID as inputs. However, we do not need this as we are using TSS, so only need the sequence number.
I refactored the code to account for this change.

File: modules/sdk-coin-atom/src/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -153,14 +155,14 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
...
-      }
-    }
+    // for (const amount of amountArray) {
+    //   if (!amount.amount) {
+    //     throw new BuildTransactionError('Invalid amount: undefined');
+    //   }
+    //   if (!amount.denom) {
+    //     throw new BuildTransactionError('Invalid denom: undefined');
+    //   }
+    // }
```
Comment (mukeshsp) [inline modules/sdk-coin-atom/src/lib/transactionBuilder.ts:165]: Please remove the commented code and add the ticket number in todo.

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 3]
```diff
@@ -203,17 +197,17 @@ export class Utils implements BaseUtils {
...
+    // @ts-ignore
+    return makeSignDoc(txBodyBytes, authInfoBytes, null, null);
   }
 
-  createSignedTxRaw(pubkey: string, signDoc: SignDoc, signature: any): TxRaw {
-    const signatureBytes = new Secp256k1Signature(fromHex(signature.r), fromHex(signature.s)).toFixedLength();
-    const stdSignature = encodeSecp256k1Signature(fromHex(pubkey), signatureBytes);
+  createSignedTxRaw(atomSignature: Signature, signDoc: SignDoc): TxRaw {
+    console.log('atom sig at this point: ' + atomSignature.signature);
+    console.log('hex of that signature: ' + atomSignature.signature.toString('hex'));
```
Comment (mukeshsp) [inline modules/sdk-coin-atom/src/lib/utils.ts:208]: remove console log

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 4]
```diff
@@ -203,17 +197,17 @@ export class Utils implements BaseUtils {
...
+    const stdSignature = encodeSecp256k1Signature(fromHex(atomSignature.publicKey.pub), atomSignature.signature);
+    console.log('stdSig: ' + stdSignature.signature);
```
Comment (mukeshsp) [inline modules/sdk-coin-atom/src/lib/utils.ts:210]: remove console

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 5]
```diff
@@ -225,6 +219,31 @@ export class Utils implements BaseUtils {
     const signedTxBytes = TxRaw.encode(signedTxRaw).finish();
     return toBase64(signedTxBytes);
   }
+
+  isSignedRawTx(rawTransaction: string): boolean {
+    // if (!this.isValidRawTransaction(rawTransaction)) {
+    //   return false;
+    // }
```
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/utils.ts:226]: Good to remove

File: modules/sdk-coin-atom/src/lib/utils.ts [hunk 6]
```diff
@@ -225,6 +219,31 @@ export class Utils implements BaseUtils {
...
+    const decodedTx = this.getDecodedTxFromRawBase64(rawTransaction);
+    if (decodedTx.authInfo.signerInfos.length > 0) {
+      return true;
+    }
+    return false;
+  }
+
+  getSignerInfoFromRawSignedTx(rawTransaction: string) {
+    if (!this.isSignedRawTx(rawTransaction)) {
```
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/utils.ts:229]: This method gets called within `fromRawTransaction` in the transaction class. Before this method is called, we already check if the transaction is indeed signed or not (Line 123 in transaction class), so we might not need line 235 (unless this method can be called from elsewhere where we need to check if the transaction is signed)

Comment (mukeshsp) [reply modules/sdk-coin-atom/src/lib/utils.ts:229]: Let's keep this function for now, as we may call it in another class in the future!

File: modules/sdk-coin-atom/package.json [hunk 7]
```diff
@@ -44,7 +44,6 @@
     "@bitgo/statics": "^14.1.0",
     "@bitgo/utxo-lib": "^7.5.0",
     "@cosmjs/amino": "^0.29.5",
-    "@cosmjs/crypto": "^0.29.5",
```
Comment (sachushaji) [inline modules/sdk-coin-atom/package.json:47]: No package lock changes?

File: modules/sdk-coin-atom/src/lib/iface.ts [hunk 8]
```diff
@@ -29,7 +29,7 @@ export interface TxData extends AtomTransaction {
 export interface AtomTransaction {
   type: string;
   signerAddress: string;
-  explicitSignerData: SignerData;
+  sequence: number;
```
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/iface.ts:32]: Seems to be breaking change?, mention in the commit message ([format](https://www.conventionalcommits.org/en/v1.0.0/#:~:text=in%20Semantic%20Versioning\).-,BREAKING,-CHANGE%3A%20a))

**Review by sachushaji — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-atom/package.json (hunk 7 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/package.json:47]: Isn't the lock-file automatically generated?
I do not see it in other coin modules
https://github.com/BitGo/BitGoJS/tree/master/modules/sdk-coin-atom
https://github.com/BitGo/BitGoJS/tree/master/modules/sdk-coin-sol

File: modules/sdk-coin-atom/src/lib/transactionBuilder.ts (hunk 2 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transactionBuilder.ts]: Marking as resolved

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 3 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts]: Marking as resolved

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 4 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts]: Marking as resolved

File: modules/sdk-coin-atom/src/lib/utils.ts (hunk 6 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts:229]: Good call - I was trying to explicitly show in this version that currently we are only supporting signed tx data, and are working on unsigned with the HSM team

File: modules/sdk-coin-atom/src/lib/iface.ts (hunk 8 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/iface.ts:32]: Updated the commit message for the breaking change commit!

**Review by mukeshsp — APPROVED** [review]

File: modules/sdk-coin-atom/package.json (hunk 7 above)
Comment (sachushaji) [reply modules/sdk-coin-atom/package.json:47]: Yeah, Lock files are autogenerated when you do `yarn`  again if you made changes to dependencies. I checkout locally and seems like it's a peer dependencies for some of the packages like `@cosmjs/amino` etc, probably explains why lock file was not modified.

**Review by sachushaji — APPROVED** [review]

File: modules/sdk-coin-atom/package.json (hunk 7 above)
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/package.json:47]: Oh yes - they're dependencys of each other, thanks for checking!

**Pull request merged**